packages/mui-joy/src/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,197 @@
+  },
+  theme.variants[ownerState.variant!]?.[ownerState.color!],
+  theme.variants[`${ownerState.variant!}Hover`]?.[ownerState.color!],
+  theme.variants[`${ownerState.variant!}Active`]?.[ownerState.color!],
+  theme.variants[`${ownerState.variant!}Disabled`]?.[ownerState.color!],
+]);
+
+const IconButton = React.forwardRef(function IconButton(inProps, ref) {
+  const props = useThemeProps<typeof inProps & { component?: React.ElementType }>({
+    props: inProps,
+    name: 'MuiIconButton',
+  });
+
+  const {
+    children,
+    className,
+    action,
+    component = 'button',
+    color = 'primary',
+    variant = 'light',
+    size = 'md',
+    shape = 'rounded',
+    ...other
+  } = props;
+
+  const buttonRef = React.useRef<HTMLElement | null>(null);
+  const handleRef = useForkRef(buttonRef, ref);
+
+  const ComponentProp = component;
+
+  const { focusVisible, setFocusVisible, getRootProps } = useButton({
+    ...props,
+    component: ComponentProp,
+    ref: handleRef,
+  });
+
+  React.useImperativeHandle(
+    action,
+    () => ({
+      focusVisible: () => {
+        setFocusVisible(true);
+        buttonRef.current?.focus();
+      },
+    }),
+    [setFocusVisible],
+  );
+
+  const ownerState = {
+    ...props,
+    component,
+    color,
+    variant,
+    size,
+    shape,
+    focusVisible,
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  return (
+    <IconButtonRoot
+      as={ComponentProp}
+      className={clsx(classes.root, className)}
+      ownerState={ownerState}
+      {...other}
+      {...getRootProps()}
+    >
+      {children}
+    </IconButtonRoot>
+  );
+}) as ExtendIconButton<IconButtonTypeMap>;
+
+IconButton.propTypes /* remove-proptypes */ = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit TypeScript types and run "yarn proptypes"  |
+  // ----------------------------------------------------------------------
+  /**
+   * A ref for imperative actions. It currently only supports `focusVisible()` action.
+   */
+  action: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({
+      current: PropTypes.shape({
+        focusVisible: PropTypes.func.isRequired,
+      }),
+    }),
+  ]),
+  /**
+   * @ignore
+   */
+  children: PropTypes.node,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
+  /**
+   * The color of the component. It supports those theme colors that make sense for this component.
+   * @default 'primary'
+   */
+  color: PropTypes.oneOf(['context', 'danger', 'info', 'neutral', 'primary', 'success', 'warning']),
+  /**
+   * The component used for the root node.
+   * Either a string to use a HTML element or a component.
+   */
+  component: PropTypes.elementType,
+  /**
+   * The shape of the component.
+   */
+  shape: PropTypes.oneOf(['circular', 'rounded', 'square']),
```

Comment:
```suggestion
  shape: PropTypes.oneOf(['circular', 'rounded', 'squared']),
```

If we use adjectives, we should be consistent